scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -0,0 +1,285 @@
+        """
+        Call get_gem_json_data
+
+        :param gem_path: Gem path to gather info for
+
+        :return dict with project info. Otherwise None is returned
+    """
+    return dict()
+
+
+def get_all_gem_infos(project_path: str) -> list:
+    """
+        Get list of all avaliable gem paths for project at project_path using get_all_gems
+
+        Gather together list of dicts for each gem using get_gem_json_data
+
+        :param project_path: Project path to gather avaliable gem infos for
+
+        :return list of dicts containing gem infos.
+    """
+    return list()
+
+
+def download_gem(gem_name: str, force_overwrite: bool = False, progress_callback = None):
+    """
+        Call download_gem
+
+        :param gem_name: Name of gem to download from a repo
+        :param force_overwrite: bool if False does not overwrite gem if already downloaded, otherwise overwrites gem
+        :param progress_callback: A cpp function callback to update progress
+    """
+    pass
+
+
+def is_gem_update_avaliable(gem_name: str, last_updated: str) -> bool:
+    """
+        Call is_o3de_gem_update_available
+
+        :param gem_name: Name of gem to check for update on
+        :param last_updated: Timestamp when gem was last updated, must be parsable using datetime.fromisoformat
+
+        :return bool True if update is avaliable, otherwise False.
+    """
+    return False
+
+
+#### Gem Repo methods ###
+
+def register_gem_repo(repo_uri: str, remove: bool = False):
+    """
+        Un/Registers gem repo using register
+
+        :param repo_uri: Uri of repo to register
+        :param remove: bool if False register repo, otherwise remove the repo from registry
+    """
+    pass
+
+
+def get_all_repo_infos() -> list:
+    """
+        Get list of all registered repo uris get_manifest_repos
+
+        Gather together list of dicts for each repo using get_repo_json_data
+        Also insert the repo path from get_repo_path as the path attr into each dict
+
+        :return list of dicts containing repo infos.
+    """
+    return list()
+
+
+def get_gem_infos_from_repo(repo_uri: str) -> list:
+    """
+        Call get_gem_json_paths_from_cached_repo for repo_uri
+
+        Gather together list of dicts for each gem using get_gem_json_data
+
+        :param repo_uri: Uri of repo to gather gemo infos from
+
+        :return list of dicts containing gem infos.
+    """
+    return list()
+
+
+def get_gem_infos_from_all_repos() -> list:
+    """
+        Call get_gem_json_paths_from_all_cached_repos
+
+        Gather together list of dicts for each gem using get_gem_json_data
+
+        :return list of dicts containing gem infos.
+    """
+    return list()
+
+
+def refresh_gem_repo(repo_uri: str):
+    """
+        Call refresh_repo with repo_uri
+
+        :param repo_uri: Uri of repo to refresh
+    """
+    pass
+
+def refresh_all_gem_repos():
+    """
+        Call refresh_repos
+    """
+    pass
```

Comment:
would we not want these calls to return the refreshed repo contents, or do we want to avoid that in case nothing has changed?  Either way we may end up having to make another call after refreshing to check if the contents are updated.

Reply:
We don't need it currently so I'll leave it as is, it can be changed it the future if we use it though.